paasta_tools/cli/cmds/wait_for_deployment.py

```diff
@@ -137,9 +138,16 @@ def validate_git_sha(git_sha, git_url, deploy_group, service):
     Raise exception when the provided git_sha is not the latest
     marked for deployment in 'deploy_group' for 'service'.
```

Comment:
This is kinda weird. I remember why I added this in there: we want to protect people from "waiting forever" on a sha that isn't marked.

But in reality, this is kinda slow and if we are not going to block on it anyway, I think maybe we don't really need this verification.